api/src/shared/infrastructure/utils/knex-utils.js

```diff
@@ -7,27 +7,38 @@ const DEFAULT_PAGINATION = {
 
 /**
  * Paginate a knex query with given page parameters
- * @param {*} queryBuilder - a knex query builder
- * @param {Object} page - page parameters
- * @param {Number} page.number - the page number to retrieve
- * @param {Number} page.size - the size of the page
+ * @param {object} params
+ * @param {object} params.queryBuilder - a knex query builder
+ * @param {object} params.paginationParams
+ * @param {Number} params.paginationParams.number - the page number to retrieve
+ * @param {Number} params.paginationParams.size - the size of the page
+ * @param {object|null} params.trx - transaction to use
+ * @param {object|null} params.countQueryBuilder - a knex query builder that counts the total number of rows, bypassing the default one
  */
-const fetchPage = async (
+const fetchPage = async ({
   queryBuilder,
-  { number = DEFAULT_PAGINATION.PAGE, size = DEFAULT_PAGINATION.PAGE_SIZE } = {},
-  countRequestBuilder = undefined,
-) => {
+  paginationParams: { number = DEFAULT_PAGINATION.PAGE, size = DEFAULT_PAGINATION.PAGE_SIZE } = {},
+  trx = null,
+  countQueryBuilder = null,
+}) => {
   const page = number < 1 ? 1 : number;
   const offset = (page - 1) * size;
 
-  const countExecutor = countRequestBuilder
-    ? countRequestBuilder
-    : knex.count('*', { as: 'rowCount' }).from(queryBuilder.clone().as('query_all_results'));
+  const countExecutor = countQueryBuilder
+    ? countQueryBuilder
+    : trx
+      ? trx.count('*', { as: 'rowCount' }).from(queryBuilder.clone().as('query_all_results'))
+      : knex.count('*', { as: 'rowCount' }).from(queryBuilder.clone().as('query_all_results'));
```

Comment:
Il ne faudrait pas gérer le cas où on a un `countQueryBuilder` ET `trx` ?

On peut écrire quelques chose du genre (à vérifier):
```suggestion
  const defaultCountQuery = knex.count('*', { as: 'rowCount' })
    .from(queryBuilder.clone().as('query_all_results'));
  
  let countExecutor = countQueryBuilder || defaultCountQuery;
  if (trx) {
    countExecutor = countExecutor.transacting(trx);
  }
```

Reply:
AH. Ah ben oui, bien vu merci